source/middlewares/extract-jwt.ts

```
@@ -0,0 +1,35 @@
+import Jwt from 'jsonwebtoken';
+import { Context } from 'koa';
+import Config from '../configs/config';
+
+
+export async function  verifyToken(
+  ctx: Context,
+  next
+) {
+
+  const bearer = ctx.get('Authorization').split(' ');
+
+  let allowed = true;
+  let decoded;
+
+  const token = bearer[1];
+
+  if (bearer[0] !== 'Bearer') {
+    allowed = false;
+  }
+
+  if (!token) {
+    allowed = false;
+  }
+  try {
+    decoded = Jwt.verify(token, Config.token.secret);
+  } catch (e) {
+    ctx.status=400
+    ctx.body='Session expired'
```

We cannot assume that the error is due to an expired token. You can throw a more generalized error like "Invalid token.".

---

Also as a suggestion, you can use `ctx.throw()` method.
https://koajs.com/#ctx-throw-status-msg-properties-